blog/2018/12/20/mojolicious-serving-angular-web-app/index.markdown

```diff
@@ -0,0 +1,379 @@
+Copy Angular app compiled into `dist/` to `public` directory of mojo app so that mojo will serve automatically. 
+
+    Sachin@02:12 PM[~/workspace/project/mojo_angular/NgDemo]$ cd dist
+    Sachin@02:13 PM[~/workspace/project/mojo_angular/NgDemo/dist]$ ls NgDemo
+    Sachin@02:13 PM[~/workspace/project/mojo_angular/NgDemo/dist]$cp -R NgDemo ~/workspace/project/mojo_angular/mojo_angular_app/public/
+    Sachin@02:13 PM[~/workspace/project/mojo_angular/NgDemo/dist]
+
+##### c. Modify Mojo Application class to serve angular app:
+Mojo Application class, `~/workspace/project/mojo_angular/mojo_angular_app/lib/MojoAngularApps.pm` in our demo, should be modified
+to push compiled angular app directory to mojo's static paths list so that it is served automatically as well.
+
+    package MojoAngularApp;
+
+    use Mojo::Base 'Mojolicious';
+
+    # This method will run once at server start       
+    sub startup {
+        my $self = shift;
+
+        # Load configuration from hash returned by "my_app.conf"                                                                         
+        my $config = $self->plugin('Config');
+        # Documentation browser under "/perldoc"                                       
+        $self->plugin('PODRenderer') if $config->{perldoc};
+
+        # Router                                          
+        my $r = $self->routes;
+    
+        # serve angular SPA located at static page location                                                                              
+        push @{$self->static->paths} => '/home/sachin/workspace/project/mojo_angular/mojo_angular_app/public/NgDemo';
+
+        # Normal route to controller                      
+        $r->get('/')->to('example#welcome');
+    }
+
+    1;
+
+Magic line above is:
+
+    push @{$self->static->paths} => '/home/sachin/workspace/project/mojo_angular/mojo_angular_app/public/NgDemo';
+
+##### d. Run hypnotoad server to see if angular page is served
+
+    Sachin@02:48 PM[~/workspace/project/mojo_angular/mojo_angular_app]$ hypnotoad -f script/mojo_angular_app 
+    [Sat Dec 15 14:49:03 2018] [info] Listening at "http://*:8080"
+    Server available at http://127.0.0.1:8080
+    [Sat Dec 15 14:49:03 2018] [info] Manager 40633 started
+    [Sat Dec 15 14:49:03 2018] [info] Creating process id file "/Users/Sachin/workspace/project/mojo_angular/mojo_angular_app/script/hypnotoad.pid"
+    [Sat Dec 15 14:49:03 2018] [info] Worker 40634 started
+    [Sat Dec 15 14:49:03 2018] [info] Worker 40635 started
+    [Sat Dec 15 14:49:03 2018] [info] Worker 40637 started
+    [Sat Dec 15 14:49:03 2018] [info] Worker 40636 started
+
+![mojolicious serving angular SPA](mojo_serving_angular.png)
+
+###### Congratulations! we have served angular app with Mojolicious. Please note that the URL is the public route of mojo app `http://localhost:8080/NgDemo/`
+
+
+## Growing App
```

Comment:
"Growing" means a specific thing in the Mojo documentation, namely moving from a lite app to a full app. Could this be called something like "Integrating the Apps"?